### PR TITLE
Merge resposnes from NLU and Domain when there're no retrieval intents

### PR DESCRIPTION
**Pull request by degiz**
**Proposed changes**:
- Make sure the `responses` are synced between NLU data and the Domain even if there're no retrieval intents in the training data.

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/shared/importers/rasa.py [hunk 1]
```diff
@@ -65,7 +65,6 @@ async def get_domain(self) -> Domain:
             return domain
         try:
             domain = Domain.load(self._domain_path)
-            domain.check_missing_templates()
```
Comment (dakshvar22) [inline rasa/shared/importers/rasa.py:68]: Shouldn't this check be done somewhere if not here?

**Review by dakshvar22 — COMMENTED** [review]
Question: Using the `RetrievalModelDataImporter` to achieve this without having any retrieval intents seems weird. Shouldn't the logic of syncing all responses defined in NLU data(irrespective of them being response templates for retrieval intents) be done inside `CombinedDataImporter` itself and then `RetrievalModelDataImporter` adds retrieval intents to domain and fetches the response templates from domain. What do you think?

File: rasa/shared/importers/rasa.py (hunk 1 above)
Comment (degiz) [reply rasa/shared/importers/rasa.py:68]: I've added it [here](https://github.com/RasaHQ/rasa/pull/7390/files#diff-663c0521028355e3657687cc698278f0d680619c6897bbe0762fe0e40213143eR328)

**Comment by dakshvar22** [comment]
@degiz I also do not have very strong opinions but renaming it more appropriately(something like - `NLUDomainSyncImporter`) seems like a better approach I think.

**Comment by degiz** [comment]
@dakshvar22 I've change the name! Do you have time to re-review?

**Review by dakshvar22 — APPROVED** [review]
Looks good. Thanks for fixing this 💯

**Pull request merged**